eth/abc.py

```diff
@@ -1688,6 +1689,22 @@ def state_in_temp_block(self) -> ContextManager[StateAPI]:
         ...
 
 
+class VirtualMachineModifierAPI(ABC):
```

Comment:
I've been wondering if this should be named something like `ConsensusEngineAPI`. That would be more aligned with how it is used but on the other hand, I'm usually a fan of small focused and therefore very generic interfaces which may be sticked together to create more specialized interfaces. And with that in mind, there's no reason why this should be called `ConsensusEngineAPI` when in reality there's nothing consensus engine specific about it.